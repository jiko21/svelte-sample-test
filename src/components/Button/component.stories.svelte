<script>
  import { Meta, Template, Story } from '@storybook/addon-svelte-csf';
  import Button from './component.svelte';
</script>

<Meta
  title="components/Button"
  component={Button}
  argTypes={{
    label: { control: 'text' },
    style: {
      type: 'select',
      options: ['primary', 'danger'],
    },
    disabled: { control: 'bool' },
    onClick: { action: 'onClick' },
  }}
/>

<Template let:args>
  <Button {...args} on:click={args.onClick} />
</Template>

<Story
  name="primary"
  args={{
    label: 'button',
    style: 'primary',
    disabled: false,
  }}
/>

<Story
  name="danger"
  args={{
    label: 'button',
    style: 'danger',
    disabled: false,
  }}
/>

<Story
  name="Disabled"
  args={{
    label: 'button',
    disabled: true,
  }}
/>
