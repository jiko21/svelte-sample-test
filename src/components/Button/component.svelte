<script lang="ts">
  type ButtonStyle = 'primary' | 'danger';
  export let label = '';
  export let style: ButtonStyle = 'primary';
  export let disabled = false;
  export let testId = '';

  let className: ButtonStyle | 'disabled' = style;
  $: computeStyle(disabled)

  const computeStyle = (disabled) => {
    className = disabled ? 'disabled' : style;
  }
</script>

<button class={className} data-testid={testId} disabled={disabled} on:click>
  {label}
</button>

<style>
  button {
    border: none;
    border-radius: 4px;
    font-weight: 700;
    height: 32px;
    line-height: 16px;
    min-width: 60px;
    padding: 8px 4px;
    position: relative;
  }

  button:active {
      top: 1px;
  }

  button.primary {
    background-color: #00c1cf;
    color: white;
  }

  button.primary:hover {
    background-color: #5ec9d1;
    color: white;
  }

  button.danger {
      background-color: #ff002b;
      color: white;
  }

  button.danger:hover {
      background-color: #ff3b3b;
      color: white;
  }

  button.disabled {
    background-color: #dfdfdf;
    color: #3c3c3c;
  }

  button.disabled:active {
      top: 0;
  }
</style>
