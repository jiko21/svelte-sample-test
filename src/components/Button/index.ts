import Button from './component.svelte';

export default Button;
