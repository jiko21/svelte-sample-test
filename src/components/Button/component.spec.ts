import { fireEvent, screen } from '@testing-library/dom';
import {render} from '@testing-library/svelte'

import Button from './component.svelte';

describe('components/Button', () => {
  const onClickMock = jest.fn();
  beforeEach(() => {
    jest.clearAllMocks();
  });
  describe('primary', () => {
    it('correctly render', () => {
      const component = render(Button, {
        label: 'button',
      });
      expect(component.container).toMatchSnapshot();
      expect(screen.getByText('button')).toBeTruthy();
    });

    it('call function correctly', () => {
      const { component } = render(Button, {
        label: 'button',
      });

      component.$on('click', onClickMock);
      fireEvent.click(screen.getByText('button'));

      expect(onClickMock).toBeCalled();
    });
  });

  describe('danger', () => {
    it('correctly render', () => {
      const component = render(Button, {
        label: 'button',
        style: 'danger',
      });
      expect(component.container).toMatchSnapshot();
      expect(screen.getByText('button')).toBeTruthy();
    });

    it('call function correctly', () => {
      const { component } = render(Button, {
        label: 'button',
        style: 'danger',
      });

      component.$on('click', onClickMock);
      fireEvent.click(screen.getByText('button'));

      expect(onClickMock).toBeCalled();
    });
  });

  describe('disabled', () => {
    it('correctly render', () => {
      const component = render(Button, {
        label: 'button',
        disabled: true,
      });
      expect(component.container).toMatchSnapshot();
      expect(screen.getByText('button')).toBeTruthy();
    });
  });
})
