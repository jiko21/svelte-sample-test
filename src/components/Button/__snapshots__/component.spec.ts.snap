// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Button danger correctly render 1`] = `
<body>
  <div>
    <button
      class="danger svelte-1ohb8zt"
      data-testid=""
    >
      button
    </button>
  </div>
</body>
`;

exports[`components/Button disabled correctly render 1`] = `
<body>
  <div>
    <button
      class="disabled svelte-1ohb8zt"
      data-testid=""
      disabled=""
    >
      button
    </button>
  </div>
</body>
`;

exports[`components/Button primary correctly render 1`] = `
<body>
  <div>
    <button
      class="primary svelte-1ohb8zt"
      data-testid=""
    >
      button
    </button>
  </div>
</body>
`;
